gtsfm/utils/sampling.py

```diff
@@ -39,3 +43,42 @@ def sample_points_on_plane(
 
     pts = np.hstack([x, y, z])
     return pts
+
+
+def sample_random_directions(num_samples: int) -> List[Unit3]:
+    """Samples `num_samples` Unit3 3D directions, uniformly in Cartesian coordinates.
+
+    Args:
+        num_samples: Number of samples required.
+
+    Returns:
+        List of sampled Unit3 directions.
+    """
+    samples = np.random.normal(size=(num_samples, 3))
+    return [Unit3(sample / np.linalg.norm(sample)) for sample in samples]
```

Comment:
will Unit3 normalize these automatically? or is there a Unit3.normalize() we can call?

Reply:
yeah it normalizes internally.

Reply:
this was written by copilot and I felt it was okay :)

Reply:
changed it now anyway